db-api/index.js

```diff
@@ -0,0 +1,106 @@
+// required capabilities:
+// - CAP_NET_BIND_SERVICE (For listening on a port < 1024 so users can't steal its port number)
+// DO NOT run as root
+if(process.getuid()==0||process.geteuid()==0) {
+  throw new Error("Refusing to run as root. This should be running as an unprivileged user.");
+}
+import express from "express";
+import identHelper from "./ident-helper.js";
+import sql from "./database.js";
+const safe_regex = /^[A-Za-z0-9_-]+$/;
+
+function get_listen_port() {
+  if (
+    process.env.PORT &&
+    parseInt(process.env.PORT) > 0 &&
+    parseInt(process.env.PORT) < 65536
+  ) {
+    return parseInt(process.env.PORT);
+  }
+}
+const app = express();
+app.use(identHelper);
+app.use((req, res, next) => {
+  if (req.username == "root") {
+    return res.status(500).json({
+      message:
+        "Refusing to respond to request from root. Try accessing this API as an unprivileged user.",
+    });
+  }
+  if (req.username.includes("_")) {
+    return res.status(500).json({
+      message: "Invalid username, contact an admin in #nest",
+    });
+  } else next();
+});
+app.use((req, res, next) => {
+  console.log(`${req.method} ${req.originalUrl} - ${req.socket.remoteAddress}:${req.socket.remotePort} ${req.username}`);
+  next();
+})
+app.get("/setup/:username", async function (req, res) {
+  if (safe_regex.test(req.params.username) !== true) {
+    return res.status(500).json({
+      message:
+        "Username contains invalid characters, see /usr/local/nest/db-api/index.js line 6 for the regex used in production.",
```

Comment:
```suggestion
        "Username contains invalid characters, see /usr/local/nest/db-api/index.js for the regex used in production.",
```
Removing line number reference since line numbers will change and we'll forget to update it.